ui/imports/shared/controls/TransactionDelegate.qml

```diff
@@ -9,48 +9,103 @@ import StatusQ.Controls 0.1
+        transactionType: TransactionDelegate.Send
+        formatCurrencyAmount: RootStore.formatCurrencyAmount
+        loading: isModelDataValid && modelData.loadingTransaction
+    }
+   \endqml
+
+   Additional usages should be handled using states.
+*/
 
 StatusListItem {
     id: root
 
-    property alias cryptoValueText: cryptoValueText
-    property alias fiatValueText: fiatValueText
+    signal retryClicked()
 
     property var modelData
     property string symbol
-    property bool isIncoming
+    property string swapSymbol // TODO fill when swap data is implemented
+    property int transactionType
+    property int transactionStatus: transferStatus === 0 ? TransactionDelegate.TransactionStatus.Failed : TransactionDelegate.TransactionStatus.Finished
     property string currentCurrency
     property int transferStatus
     property double cryptoValue
+    property double swapCryptoValue // TODO fill when swap data is implemented
     property double fiatValue
+    property double swapFiatValue // TODO fill when swap data is implemented
+    property double feeCryptoValue // TODO fill when bridge data is implemented
+    property double feeFiatValue // TODO fill when bridge data is implemented
     property string networkIcon
     property string networkColor
     property string networkName
-    property string shortTimeStamp
+    property string bridgeNetworkName // TODO fill when bridge data is implemented
+    property string timeStampText
     property string savedAddressNameTo
     property string savedAddressNameFrom
-    property bool isSummary: false
+    property var formatCurrencyAmount: function() {}
 
     readonly property bool isModelDataValid: modelData !== undefined && !!modelData
     readonly property bool isNFT: isModelDataValid && modelData.isNFT
-    readonly property string name: isModelDataValid ?
-                                        root.isNFT ?
-                                            modelData.nftName ?
-                                                modelData.nftName :
-                                                "#" + modelData.tokenID :
-                                            root.isSummary ? root.symbol : RootStore.formatCurrencyAmount(cryptoValue, symbol) :
-                                        "N/A"
-
-    readonly property string image: isModelDataValid ?
-                                        root.isNFT ?
-                                            modelData.nftImageUrl ?
-                                                modelData.nftImageUrl :
-                                                "" :
-                                            root.symbol ?
-                                                Style.png("tokens/%1".arg(root.symbol)) :
-                                                "" :
-                                        ""
+    readonly property string transactionValue: {
+        if (!isModelDataValid)
+            return "N/A"
```

Comment:
Is this an international way of showing missing data or it is translatable?

Reply:
Good point, it's only for English, added `qsTr`